### ISSUE TITLE
E2E + Unit Testing + ?

### ISSUE THREAD

**Issue by pjurczynski** [pjurczynski/inmate_kata#2]
I think that if we want to implement a more complex, more real-life example, then we should also add some e2e tests.

Thoughts?

**Comment by Greegko** [comment]
Yeah I think we can add, but maybe after we are going though a bit more advanced and have 2 small components where we should check the integration. For just a small function E2E seems overkill.

**Comment by Greegko** [comment]
How would you like to start UI? After we finish the functionality or try to do it pararell?

**Comment by pjurczynski** [comment]
> Yeah I think we can add, but maybe after we are going though a bit more advanced and have 2 small components where we should check the integration. For just a small function E2E seems overkill.

TLDR;
let's drop e2e and other testing - let's focus on simple unit tests for now

Long version:
I thought about trying to emulate a real world scenario, where you want to implement an app/feature. Then this doesn't really start with "I want to implement just a small function that adds 2 numbers". You start more with "I want a calculator UI that can do additions". I'm super interested if we could pull this off ;). 

But we can also try and start with something small to see if this concept of "inmate_kata" would even make sense.

> How would you like to start UI? After we finish the functionality or try to do it pararel?
> From my side we can do the both, and we put in the rule, that we cannot make 2 tests / commits in the same things (Unit / E2E) after each other by the same user.

I would say let's start with the functionality. Let's peek a simple problem and testout the framework.

For next problem, I would like to tackle something bigger, that would maybe also require to write some "ticket" or baseline requirements.

**Comment by Greegko** [comment]
For me it is fine, we have two kind of development, one functionality and one for UI. In this case we can practice how to write tests when the functionality is not done yet, but still we need to create something. Maybe it is not possible, but at least we have the option to question it and look for options.
